sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py

```diff
@@ -2,83 +2,129 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from typing import Any, TYPE_CHECKING
+from typing import Any, Union, TYPE_CHECKING
 
 from azure.core.configuration import Configuration
 from azure.core.pipeline import AsyncPipeline
-from azure.core.pipeline.policies import(
-    ContentDecodePolicy, UserAgentPolicy, DistributedTracingPolicy, HttpLoggingPolicy
+from azure.core.pipeline.policies import (
+    ContentDecodePolicy,
+    UserAgentPolicy,
+    DistributedTracingPolicy,
+    HttpLoggingPolicy,
 )
 from azure.core.pipeline.transport import AsyncHttpTransport
-
-from ._generated import KeyVaultClient
+from ._generated.v7_0.version import VERSION as V7_0_VERSION
+from ._generated.v2016_10_01.version import VERSION as V2016_10_01_VERSION
 from . import AsyncChallengeAuthPolicy
 from .._user_agent import USER_AGENT
 
 
 if TYPE_CHECKING:
     try:
         # pylint:disable=unused-import
-        from azure.core.credentials import TokenCredential
+        from azure.core.credentials_async import AsyncTokenCredential
+        from ._generated.v7_0 import KeyVaultClient as KeyVaultClientV7
+        from ._generated.v2016_10_01 import KeyVaultClient as KeyVaultClientV2016_10_01
+
+        KeyVaultClientTypes = Union[KeyVaultClientV7, KeyVaultClientV2016_10_01]
     except ImportError:
-        # TokenCredential is a typing_extensions.Protocol; we don't depend on that package
+        # AsyncTokenCredential is a typing_extensions.Protocol; we don't depend on that package
         pass
 
-
-class AsyncKeyVaultClientBase:
+DEFAULT_API_VERSION = V7_0_VERSION
+
+def _get_configuration_class(api_version: str) -> Configuration:
+    """
+    Get the versioned configuration implementation corresponding to the current profile.
+    :return: The versioned configuration implementation.
+    """
+    if api_version == V7_0_VERSION:
+        from ._generated.v7_0.aio._configuration_async import KeyVaultClientConfiguration as ImplConfig
+    elif api_version == V2016_10_01_VERSION:
+        from ._generated.v2016_10_01.aio._configuration_async import KeyVaultClientConfiguration as ImplConfig
+    else:
+        raise NotImplementedError(
+            f"API version {api_version} is not supported by this package. "
+            + f"We currently support versions {V7_0_VERSION} and {V2016_10_01_VERSION}"
```

Comment:
Nitpicking the error again: "we" seems weird to me. How about "Supported versions: ..."?